src/components/dialogs/parameters/sensi/sensitivity-analysis-parameters.js

```diff
@@ -192,21 +214,120 @@ export const SensitivityAnalysisParameters = ({
+            sensitivityHVDC: values.sensitivityHVDC
+                .filter((entry) => entry[ACTIVATED])
+                .map((entry) => entry[COUNT])
+                .reduce((a, b) => a + b, 0),
+            sensitivityPST: values.sensitivityPST
+                .filter((entry) => entry[ACTIVATED])
+                .filter((entry) => entry[COUNT])
+                .reduce((a, b) => a + b, 0),
+        };
+        return Object.values(resultCountByTab).reduce((a, b) => a + b, 0);
+    }, [getValues]);
+
+    const hasFormChanged = useCallback(
+        (isFormChanged) => {
+            isFormChanged && setAnalysisComputeComplexity(getResultCount());
+        },
+        [setAnalysisComputeComplexity, getResultCount]
+    );
+
+    const onChangeParams = useCallback(
+        (row, arrayFormName, index) => {
+            getSensitivityAnalysisParametersComputingCount(
+                studyUuid,
+                arrayFormName === SENSI_INJECTIONS_SET,
+                formatFilteredParams(row)
+            )
+                .then((response) => {
+                    response.text().then((value) => {
+                        setValue(
+                            `${arrayFormName}[${index}].[${COUNT}]`,
+                            value && Number(value)
+                        );
+                        setAnalysisComputeComplexity(getResultCount());
+                        hasFormChanged(false);
+                    });
+                })
+                .catch((error) => {
+                    snackError({
+                        messageTxt: error.message,
+                        headerId:
+                            'SensitivityAnalysisFilteredActiveParametersError',
+                    });
+                });
+        },
+        [
+            snackError,
+            studyUuid,
+            formatFilteredParams,
+            setValue,
+            getResultCount,
+            hasFormChanged,
+        ]
+    );
+
+    const initRowsCount = useCallback(() => {
+        const values = getValues();
+        values[PARAMETER_SENSI_INJECTIONS_SET].filter(
+            (entry) => entry[ACTIVATED] && !entry[COUNT]
+        ).forEach((entry, index) =>
+            onChangeParams(entry, PARAMETER_SENSI_INJECTIONS_SET, index)
+        );
+        values[PARAMETER_SENSI_INJECTION].filter(
+            (entry) => entry[ACTIVATED] && !entry[COUNT]
+        ).forEach((entry, index) =>
+            onChangeParams(entry, PARAMETER_SENSI_INJECTION, index)
+        );
+        values[PARAMETER_SENSI_HVDC].filter(
+            (entry) => entry[ACTIVATED] && !entry[COUNT]
+        ).forEach((entry, index) =>
+            onChangeParams(entry, PARAMETER_SENSI_HVDC, index)
+        );
+        values[PARAMETER_SENSI_PST].filter(
+            (entry) => entry[ACTIVATED] && !entry[COUNT]
+        ).forEach((entry, index) =>
+            onChangeParams(entry, PARAMETER_SENSI_PST, index)
+        );
+    }, [onChangeParams, getValues]);
```

Comment:
The same thing here too much  duplication. The only thing that changes is the  values[***]  and the second argument of the onChangeParams so you can do more refacto.